### PR TITLE
fix: adding flag to skip prompting for variables

### PR DESCRIPTION
**Pull request by duvanmonsa**
Solving issue https://github.com/stateful/runme/issues/297

Introducing new `runme run --skip-prompts` to skip prompting for variables
you can also do `runme run --skip-prompts=false` `runme run --skip-prompts=true`

### CODE REVIEW

File: internal/cmd/run.go [hunk 1]
```diff
@@ -287,6 +290,7 @@ func runCmd() *cobra.Command {
 	cmd.Flags().StringArrayVarP(&replaceScripts, "replace", "r", nil, "Replace instructions using sed.")
 	cmd.Flags().BoolVarP(&parallel, "parallel", "p", false, "Run tasks in parallel.")
 	cmd.Flags().BoolVarP(&runAll, "all", "a", false, "Run all commands.")
+	cmd.Flags().BoolVarP(&skipPromptVars, "skip-prompt-vars", "v", false, "Skip prompting for variables.")
```
Comment (duvanmonsa) [inline internal/cmd/run.go:293]: @sourishkrout I am not sure what alias should we use for this right now `v`

**Comment by duvanmonsa** [comment]
@sourishkrout Also wondering if this is something we need to do on TUI as well?

File: internal/cmd/run.go (hunk 1 above)
Comment (sourishkrout) [reply internal/cmd/run.go:293]: `v` is very common for `verbose` so let's not use it, please. How about we rename it to `--skip-prompts` with `s` as alias the `vars` part is redundant.

File: internal/cmd/run.go [hunk 2]
```diff
@@ -172,9 +173,11 @@ func runCmd() *cobra.Command {
 				return err
 			}
 
-			err = promptEnvVars(cmd, sessionEnvs, runBlocks...)
-			if err != nil {
-				return err
+			if isTerminal(os.Stdout.Fd()) && !skipPromptVars {
```
Comment (sourishkrout) [inline internal/cmd/run.go:176]: Wondering if `skipPromptVars` should be set to true if it's not a terminal and the flag's not changed explicitly. That way one could always enforce behavior with `--skip-prompts=true` or `--skip-prompts=false`.

**Review by sourishkrout — CHANGES_REQUESTED** [review]
See my two comments please @duvanmonsa.

**Comment by sourishkrout** [comment]
> @sourishkrout Also wondering if this is something we need to do on TUI as well?

That's separate. The main desire for this flag is to be used in non-interactive use cases. The TUI is inherently interactive.

File: internal/cmd/run.go (hunk 1 above)
Comment (duvanmonsa) [reply internal/cmd/run.go:293]: we are removing the alias only `--skip-prompts` will be available.

File: internal/cmd/run.go (hunk 2 above)
Comment (sourishkrout) [reply internal/cmd/run.go]: What about ☝️ @duvanmonsa ?

Comment (duvanmonsa) [reply internal/cmd/run.go]: I changed the default value to `true` @sourishkrout

**Comment by sourishkrout** [comment]
```
runme run --filename README.md --all --skip-prompts
                         
Run all 24 tasks? [Y/n] Y
```

We should probably skip this prompt too @duvanmonsa, no?

**Comment by duvanmonsa** [comment]
> 

@sourishkrout my position is kind of divided right now, I get why you think we should skip it based on the flag name, but also I feel we should actually ask for that confirmation it is up to you.

**Comment by sourishkrout** [comment]
let's skip it too @duvanmonsa. I feel like `--skip-prompts` just do what it says especially in non-TTY places like C/I.

**Comment by duvanmonsa** [comment]
Ok changes to skip that confirmation have been done as well!

**Review by sourishkrout — APPROVED** [review]
👍 LGTM

**Pull request merged**